<a href="https://colab.research.google.com/github/Tranminhtuan48/pythonbasics/blob/main/BaiTap10_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Thí  dụ : Han có 500 bài hát trong máy điện thoại. Han muốn ước lượng có mấy bài hát của  mình. Han lấy mẫu 50 bài hát bất kỳ. Trong đó có 20 bài hát của Han. ví dụ mức tin cậy 99%, khoảng tin cậy cho tỷ lệ bài hát của Han là?



In [ ]:
from scipy import stats
from math import sqrt

In [ ]:
# dữ liệu mẫu(n>= 30)
Dl_Mau = 50
# Tỉ lệ  mẫu
Tl_Mau = 20/Dl_Mau
# Độ tin cậy
M_Tin_Cay = 0.99
# Độ lệch chuẩn của tỷ lệ mẫu
Dl_Chuan = sqrt((Tl_Mau * (1 - Tl_Mau))/50)
print("Do lech  chuan  cua  mauu: ", Dl_Chuan)
#Chỉ số  Z-scores
Cs_Z = (1 - M_Tin_Cay)/2
print("Xac  xuat  tren: ", Cs_Z)
# Tính z score bằng diện tích dưới đường cong (xác xuất dữ liệu trong khoảng giá trị của phân phối chuẩn)
# ppf là hàm dùng tính diện tích dưới đường cong từ z-score
z =  stats.norm.ppf(1 - Cs_Z)
print("Z score: ", z)
# Tính giới hạn lỗi 
GH_Loi = z * Dl_Chuan
print("Gioi han  loi: ",GH_Loi) 
# Tính khoảng tin cậy
K_Tin_Cay = (Tl_Mau - GH_Loi, Tl_Mau + GH_Loi)
print("Khoang  tin  cay: ", K_Tin_Cay)
# Kiểm tra khoảng tin cậy
K_Tin_Cay_Check = stats.norm.interval(K_Tin_Cay, loc = Tl_Mau, scale=Dl_Chuan)
print("Kiem Tra  Khoang  Tin  Cay: ", K_Tin_Cay_Check)

Do lech  chuan  cua  mauu:  0.06928203230275509
Xac  xuat  tren:  0.0050000000000000044
Z score:  2.5758293035489004
Gioi han  loi:  0.17845868901485806
Khoang  tin  cay:  (0.22154131098514196, 0.5784586890148581)
Kiem Tra  Khoang  Tin  Cay:  (array([0.38050901, 0.34431456]), array([0.41949099, 0.45568544]))


Với khoảng tin cậy (0.22, 0.58). Ta thấy 99% là số bài hát được hát bởi ca sĩ Han trong điện thoại của Han có tỷ lệ từ 22% đến 58%.